patchwork-capabilities/src/main/java/com/patchworkmc/mixin/capability/ClientWorldMixin.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.patchworkmc.mixin.capability;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.client.world.ClientWorld;
+
+import com.patchworkmc.impl.capability.CapabilityProviderHolder;
+
+// World implements CapabilityProviderHolder already because of WorldMixin
+@Mixin(ClientWorld.class)
+public abstract class ClientWorldMixin implements CapabilityProviderHolder {
+	@Inject(method = "<init>", at = @At("TAIL"))
+	private void initializeCapabilities(CallbackInfo callbackInfo) {
+		// TODO: Requires Dimension API (IForgeDimension)
```

Comment:
I don't think the dimension API is strictly required here, we can just implement our own initCapabilities method outside of the interface that does the same thing. In fact, this could create a circular dependency situation if we relied on the implementation in -extensions. This issue does need a bit of discussion.

Reply:
It's not *required*, as the required method returns null by default.

Reply:
So is this purely something for custom dimensions to use?

Reply:
Yes. Same thing for the ItemStack mixin.